### PR TITLE
Fixed #25920 -- Added number formatting support for locales that use non-uniform digit grouping.

### PR DESCRIPTION
**Pull request by jasisz**
https://code.djangoproject.com/ticket/25920

### CODE REVIEW

**Comment by timgraham** [comment]
buildbot, add to whitelist.

**Comment by timgraham** [comment]
This is missing a mention in the 1.11 release notes as well as a `.. versionchanged:: 1.11` annotation as described at https://docs.djangoproject.com/en/dev/internals/contributing/writing-documentation/#documenting-new-features

File: docs/ref/settings.txt [hunk 1]
```diff
@@ -1927,6 +1927,17 @@ no grouping will be applied to the number. If this setting is greater than
 ``0``, then :setting:`THOUSAND_SEPARATOR` will be used as the separator between
 those groups.
 
+For some locales there is a need to use a varying digit grouping,
```
Comment (timgraham) [inline docs/ref/settings.txt]: Minor edits:
```
Some locales use non-uniform digit grouping, e.g. ``10,00,00,000`` in
``en_IN``. For this case, you can provide a sequence with the number of digit
group sizes to be applied. The first number defines the size of the group
preceding the decimal delimiter and each number that follows defines the size
of preceding groups. If the sequence is terminated with ``-1``, no further
grouping is performed. If the sequence terminates with a ``0``, the last group
size is used for the remainder of the number.
```

**Comment by timgraham** [comment]
@claudep, does the proposed API look good?

**Comment by claudep** [comment]
I think so. @jasisz I suppose you followed the POSIX behavior (http://lh.2xlibre.net/values/grouping/)?

**Comment by jasisz** [comment]
@claudep I've followed POSIX behaviour, but in the same way python does that.
Python in `locale` describes a slightly different convention, see https://docs.python.org/2/library/locale.html#locale.localeconv - LC_NUMERIC/grouping - the difference is usage of `0` as the last number meaning "group by last previous number to the end". It will work both ways `(3, 2)` or `(3, 2, 0)`, but in the documentation I've followed python convention.

**Comment by jasisz** [comment]
@timgraham I've done the changes.

**Comment by jasisz** [comment]
@timgraham just pinging you, do I need to change anything?

**Comment by timgraham** [comment]
merged in b5a1c3a6f50362b57603e1833e44bff5628dde3c, thanks!

**Pull request closed without merging**